### ISSUE TITLE
Files stored on network locations (Google Drive) cannot be opened

### ISSUE THREAD

**Issue by infra-d** [notepad-plus-plus/notepad-plus-plus#11168]
### Description of the Issue
Files stored on network locations (Google Drive) cannot be opened.

### Steps to Reproduce the Issue

1.  Save a txt file in G Drive
2.  Open this file from the Windows explorer.  Google drive must be installed in Windows (mapped as local drive, Eg: G:)

### Expected Behavior
Should be able to open the file from remote network location

### Actual Behavior
Error: Cannot open file

### Debug Information
Notepad++ v8.3   (64-bit)
Build time : Feb  3 2022 - 03:39:05
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "G:\My Drive\xxxx\My Documents\xxxxx\mail.txt"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19042.1466
Current ANSI codepage : 932
Plugins : CsvQuery.dll Explorer.dll MarkdownViewerPlusPlus.dll mimeTools.dll NppConverter.dll NppExport.dll NppToolBucket.dll

**Comment by dfs-** [comment]
Actual network locations work without any issues here. Tried regular network drives as well as UPnP drives and even OneDrive files.
I'd suspect it's a problem with Google Drive.

**Comment by infra-d** [comment]
My G drive is related to my company's account, hence I cannot share any detail.

The issue started about 3 weeks ago, since update to 8.2.1.

I can open any files from my G drive via the windows explorer, xlsx, docx, etc...
I can also open any txt and csv files on the network drive with Notepad.exe

Is there any other information I can provide to support this issue?

**Comment by infra-d** [comment]
Actually this issue does not happen only on G Drive but on any Network location

**Comment by Wuehli** [comment]
Hi,

i'm facing similar issues, i can't open most files from gdrive (there are only very few files that open but i can't find any logic behind), i didn't even get an error the file just won't open in notepad++.
Every other application i tried will open the files just fine.


Notepad++ v8.3.1   (64-bit)
Build time : Feb 13 2022 - 15:41:12
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "\\?\G:\Geteilte Ablagen\xxx\20220211_113523-58.hl7"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19042.1110
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll

**Comment by dfs-** [comment]
The issue might be related to the command line: 

> Command Line : "\?\G:\Geteilte Ablagen\xxx\20220211_113523-58.hl7"

Looks like the `\?` creates the issue here.

**Comment by Wuehli** [comment]
What i've posted is exactly what i got from "Debug-Info". 
This is also one of the rare files that can be opend from gdrive. 

If i should post any further infos please let me know how to find them. Thanks!

**Comment by Wuehli** [comment]
Additional Info:

When i "open" a file that wan't load in np++ the following command line will be shown in Debug:

Command Line : "\\?\G:\Geteilte Ablagen\xxx\20220214_101411-192.hl7"

**Comment by Wuehli** [comment]
corrected answere:

if a file can't be load i see this command line in Debug-Info "\\\\?\G:\Geteilte Ablagen\xxx\20220211_113523-58.hl7"

if a file can be load i see this command line in Debug-Info
Command Line : "G:\Geteilte Ablagen\xxx\20220218_134127-20.hl7"

**Comment by dfs-** [comment]
OK, so then Notepad++ does not support the Win32 File Namespace right now, could be considered as a bug.
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN#win32-file-namespaces

**Comment by Wuehli** [comment]
I didn't think the path length was the problem, so I checked it. What I can see is that files that cannot be opened have a path length >256, all other files have a path length below 256. regardless of whether the files are available offline or not.

**Comment by hoholee12** [comment]
i encountered this issue too. the workaround is to use right-click context menu to open it.

**Comment by xomx** [comment]
@Wuehli 
> files that cannot be opened have a path length >256

Yes, N++ currently cannot handle filenames longer than the `MAX_PATH` relic.
But IMO this is a different issue than others have here.

@hoholee12 @infra-d 
> Actual Behavior
>
> Error: Cannot open file

Is there any error message visible or the file opening just does nothing?

And could you try the same with your AV (MS Defender etc...) real-time protection switched temporarily off (or you can add N++ to the list of exclusions)? 

![MS-AV-off](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/13075183/a7dc38b9-236d-43ef-b536-cf2eb8b226aa)

Also if you are using the "Controlled folder access" feature (in the Ransomware Protection), add N++ to the "allowed apps".

![MS-ControlledFolderAccess](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/13075183/550561d0-2aaa-4d18-a1a0-cc71dc2d98a8)
